### PR TITLE
#166265831 set up API documentation with swagger

### PR DESCRIPTION
**Pull request by OvieMudi**
#### What does this PR do?
Adds API documentation to the project

#### Description of Task to be completed?
Ensure the API is documented correctly when accessing the endpoint `/api/v1/api-docs`

#### How should this be manually tested?
To test manually on your local computer, _run the following commands_.

```
# Clone this repository
$ git clone https://github.com/emp-daisy/HelloBooksLib.git

# Navigate to the application folder
$ cd HelloBooksLib

# Checkout the branch
$ git checkout ch/166265831/setup-api-documentation-with-swagger

# Install dependencies
$ npm install

# Start the app in development mode
$ npm run dev
```

- make a GET request to  ` localhost:3000/api/v1/api-docs`
- ensure the documentation is displayed correctly

NB: see screenshot below on how the layout should look
#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#166265831](https://www.pivotaltracker.com/story/show/166265831 "Api docs Pivotal Tracker Story")

#### Screenshots
swagger_doc4ed
swagger_doc2

### CODE REVIEW

**Review by Hadeneekeh — DISMISSED** [review]

**Review by ocranbillions — DISMISSED** [review]

**Review by chuxmykel — CHANGES_REQUESTED** [review]
I think since we're versioning our API, the docs endpoint should be /api/v1/docs.
And then, are the dummy endpoints really necessary? Let's make sure the documentation is actually working by fixing the 2 endpoints we have. And then we can build from there as we add more endpoints

**Comment by OvieMudi** [comment]
@chuxmykel my task was to make an API doc template. The dummy endpoints can be edited or replaced when a team member is documenting their feature. Let's hear what other members think tho. 

BTW, I think we only have one working endpoint for now: `/api/v1`

I agree with you on the versioning, I'll start working on this ASAP

**Comment by chuxmykel** [comment]
All right bro. Carry on

**Review by PelumiAlesh — DISMISSED** [review]

**Review by chuxmykel — APPROVED** [review]
LGTM

File: server/docs/hellobooks_api_doc.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+      parameters:
+      - name: userId
+        in: path
+        description: ID of user that needs to be updated
+        required: true
+        schema:
+          type: integer
+          format: int64
+      responses:
+        405:
```
Comment (cvjude) [inline server/docs/hellobooks_api_doc.yaml:133]: Why A 405,
It's supposed to mean Method not found.

**Review by PelumiAlesh — APPROVED** [review]

File: server/docs/hellobooks_api_doc.yaml (hunk 1 above)
Comment (OvieMudi) [reply server/docs/hellobooks_api_doc.yaml:133]: This is just a template. It is not actually doing anything yet

Comment (OvieMudi) [reply server/docs/hellobooks_api_doc.yaml:133]: When we start documenting out individual features, these will be replaced with specific error codes and messages.

**Pull request merged**